app/assets/javascripts/components/materials_bin/_fetch_data_mixin.js.coffee

```diff
@@ -0,0 +1,39 @@
+# Fetches remote data when component is visible (when @props.visible == true).
+# Data is saved under @state[@dataStateKey].
+# Client class has to define:
+#  - .dataUrl property (string)
+#  - .dataStateKey property (string), name of the state key under which data is saved
+# but it can also define:
+#  - .requestParams property (hash), argument of jQuery.ajax
+#  - .processData() (method), it can process raw AJAX response before state is updated
+window.MBFetchDataMixin =
+  getInitialState: ->
```

Comment:
You are correct.  In early versions of React there were several methods that could not be mixed in - it seems that is no longer the case (with the exception of render).  Thanks for updating me!